expan/core/statistics.py

```diff
@@ -1,83 +1,78 @@
-
-    Computation is done in form of treatment minus control, i.e. x-y
-
-    Args:
-        x (array_like): sample of a treatment group
-        y (array_like): sample of a control group
-        assume_normal (boolean): specifies whether normal distribution
-            assumptions can be made
-        percentiles (list): list of percentile values for confidence bounds
-        min_observations (integer): minimum number of observations needed
-        nruns (integer): only used if assume normal is false
-        relative (boolean): if relative==True, then the values will be returned
+def delta(x, y, assume_normal=True, alpha=0.05, percentiles=[2.5, 97.5],
+          min_observations=20, nruns=10000, relative=False):
+    """ Calculates the difference of means between the samples in a statistical sense.
+    Computation is done in form of treatment minus control, i.e. x-y.
+    Note that NaNs are treated as if they do not exist in the data. 
+    
+    :param x: sample of the treatment group
+    :type  x: pd.Series or array-like
+    :param y: sample of the control group
+    :type  y: pd.Series or array-like
+    :param assume_normal: specifies whether normal distribution assumptions can be made
+    :type  assume_normal: boolean
+    :param alpha: significance level (alpha)
+    :type  alpha: float
+    :param percentiles: list of percentile values for confidence bounds
+    :type  percentiles: list
+    :param min_observations: minimum number of observations needed
+    :type  min_observations: int
+    :param nruns: only used if assume normal is false
+    :type  nruns: int
+    :param relative: if relative==True, then the values will be returned
             as distances below and above the mean, respectively, rather than the
             absolute values. In	this case, the interval is mean-ret_val[0] to
             mean+ret_val[1]. This is more useful in many situations because it
             corresponds with the sem() and std() functions.
-        x_weights (list): weights for the x vector, in order to calculate
-            the weighted mean and confidence intervals, which is equivalent
-            to the overall metric. This weighted approach is only relevant
-            for ratios.
-        y_weights (list): weights for the y vector, in order to calculate
-            the weighted mean and confidence intervals, which is equivalent
-            to the overall metric. This weighted approach is only relevant
-            for ratios.
-        multi_test_correction (boolean): flag of whether the correction for multiple testing is needed.
-        num_tests (integer): number of tests or reported kpis used for multiple correction.
-
-    Returns:
-        DeltaStatistics object
+    :type: relative: boolean
+    
+    :return: results of type SimpleTestStatistics
+    :rtype: SimpleTestStatistics
     """
-    # Checking if data was provided
+    # Check if data was provided and it has correct format
     if x is None or y is None:
```

Comment:
Those checks are made obsolete by the later ones

Reply:
Sure!

Reply:
That's true... Well, still, I believe there is more advantages to keep it.

It might covers some tricky usage (e.g. open source user uses ```statistics.delta()``` directly). And the runtime increase to remove it is almost nothing.